arch/lkl/kernel/syscalls.c

```diff
@@ -118,6 +118,12 @@ static void del_host_task(void *arg)
 
 struct lkl_tls_key *task_key;
 
+/* Record the exit status after shutdown */
+_Atomic(int) lkl_exit_status;
```

Comment:
Exit status of what?  Presumably the first process that runs?  We should probably be tracking that outside of LKL by intercepting the exit thread group syscall.

Reply:
Right, but let's make this a separate PR.

Reply:
In that case, please can you then take ownership of factoring out our LKL history into things that are suitable for going upstream?

Reply:
Sure, but this can only be done once we have the final design for the LKL shutdown sequence. I think that this PR improves stability quite substantially, but it is not yet the final design that would be ready for upstream. There may also be a dependency with the signal handling changes.